android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/SelectLanguageFragment.java

```diff
@@ -190,6 +190,41 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
     listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
 
+      /**
+       * Utility to modify keyboardList. Based on BCP47.toggleLanguage
+       * If keyboard's languageID exists in the list, remove the keyboard.
+       * Otherwise, add the keyboard to the list.
+       * @param keyboardList
+       * @param k
+       */
+      private void toggleLanguage(ArrayList<Keyboard> keyboardList, Keyboard k) {
+        if (keyboardList != null) {
+          if (keyboardList.size() == 0) {
+            k.setLanguage(k.getLanguageID().toLowerCase(), k.getLanguageName());
+            keyboardList.add(k);
+            return;
+          }
+
+          boolean languageExists = false;
+          int index = 0;
+          // See if languageID already exists in the keyboardList
+          for (Keyboard l: keyboardList) {
+            if (BCP47.languageEquals(l.getLanguageID(), k.getLanguageID())) {
+              languageExists = true;
+              break;
+            }
+            index++;
+          }
+
+          if (languageExists && index < keyboardList.size()) {
+            keyboardList.remove(index);
+          } else {
+            k.setLanguage(k.getLanguageID().toLowerCase(), k.getLanguageName());
+            keyboardList.add(k);
+          }
```

Comment:
```suggestion
          // See if languageID already exists in the keyboardList
          for (Keyboard l: keyboardList) {
            if (BCP47.languageEquals(l.getLanguageID(), k.getLanguageID())) {
              keyboardList.remove(l);
              return;
            }
          }

          k.setLanguage(k.getLanguageID().toLowerCase(), k.getLanguageName());
          keyboardList.add(k);
```

I think this removes the need for the variables which makes it easier to read and maintain.